lite/src/skipping.rs

```diff
@@ -0,0 +1,94 @@
+use crate::types::*;
+
+/// SkippingVerifier tries to skip headers by verifying if
+/// +1/3 of the last validators it trusts has signed the new header.
+struct SkippingVerifier<H, V>
+where
+    H: Header,
+    V: ValidatorsLookup,
+{
+    trusting_period: Time,
+    state: TrustedState<H, V>,
+}
+
+impl<H, V> SkippingVerifier<H, V>
+where
+    H: Header,
+    V: ValidatorsLookup,
+{
+    /// Returns the time after which the current state expires
+    /// and the verifier must be reset subjectively.
+    fn expires(&self) -> Time {
+        self.state.header.bft_time() + self.trusting_period
+    }
+
+    /// Verify takes a header, a commit for the header,
+    /// and the next validator set referenced by the header.
+    /// Note we do not know the correct validator set for this commit -
```

Comment:
We still have a list of validators for this commit and even if we don't trust them yet IMO we should still perform the basic validation (similar to the one done for sequential). Or it can be done once we know at least one correct validator has signed.

Reply:
Agree with Anca. We still need to verify that commit is valid with respect to its validator set. If +2/3 of validators from this set signed correctly and it is confirmed by +1/3 from the trusted val set, then it's fine.

Reply:
Not sure I understand. According to the spec, we don't verify the commit according to +2/3 of its actual validator set, but only according to +1/3 of our last trusted validator set. You're saying we should do both ? Presumably we would do the +1/3 of our trusted set first, and if that checks out, then we would do the +2/3 of the actual set. But if the later check failed, it would mean that one of our trusted validators is misbehaving ...

Seems like we may want to go back to the spec on this, unless I'm understanding something wrong

Reply:
the +1/3rd check is just to decide if we should trust the actual vals in skipping case. In the sequential case we don't need to do this check. But in both cases we should check +2/3rds of actual vals signed + all the other checks (hashes, proper content for header, votes, etc). 
The meaning of the <2/3rds signing should be the same for both cases.

Reply:
> Seems like we may want to go back to the spec on this, ...

agreed.

Reply:
It seems that's not clear enough. In my head it should be as Anca said. Hopefully TLA+ will minimise this kind of comments :)

Reply:
This is obviously necessary lol. Done.